x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go

```diff
@@ -89,12 +91,43 @@ func newDiagnosticsCollectCommandWithArgs(_ []string, streams *cli.IOStreams) *c
 				return fmt.Errorf("unsupported output: %s", output)
 			}
 
-			return diagnosticsCollectCmd(streams, file, output)
+			pprof, _ := c.Flags().GetBool("pprof")
+			d, _ := c.Flags().GetDuration("pprof-duration")
+
+			return diagnosticsCollectCmd(streams, file, output, pprof, d)
 		},
 	}
 
 	cmd.Flags().StringP("file", "f", "", "name of the output diagnostics zip archive")
 	cmd.Flags().String("output", "yaml", "Output the collected information in either json, or yaml (default: yaml)") // replace output flag with different options
+	cmd.Flags().Bool("pprof", false, "Collect all pprof data from all running applications.")
+	cmd.Flags().Duration("pprof-duration", time.Second*30, "The duration to collect trace and profiling data from the debug/pprof endpoints. (default: 30s)")
+
+	return cmd
+}
+
+func newDiagnosticsPprofCommandWithArgs(_ []string, streams *cli.IOStreams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "pprof",
+		Short: "Collect pprof information from a running process.",
+		Long:  "Collect pprof information from the elastic-agent or one of its processes and write to stdout or a file.",
```

Comment:
Is my understanding correct that by default this command will take 30s to execute? Maybe we should mention this in the docs here that it is tied to pprof-duration?